File: clean.js
const fs = require('fs');
const path = require('path');

let delCount = 0;

const ignoreFiles = new Set(['.git', '.vscode', '.gitignore']);
const removeFileRule = ['cmake_install.cmake', 'CMakeCache.txt', 'CMakeFiles' , 'Makefile'];

const clean = (dirPath) => {
    // 读取目录中的所有文件和子目录
    fs.readdirSync(dirPath).forEach(file => {
        // 获取文件/子目录的完整路径
        const fullPath = path.join(dirPath, file);

        // 是否是忽略文件
        if (ignoreFiles.has(file)) {
            return;
        }

        let delFunc = fs.unlinkSync;

        // 检查该路径是文件还是目录
        const isDir = fs.statSync(fullPath).isDirectory()
        if (isDir) {
            // 递归遍历该目录
            clean(fullPath);
            delFunc = fs.rmdirSync;
        }

        // 是否符合删除规则
        removeFileRule.forEach(rule => {
            const regex = new RegExp(rule)
            if (regex.test(file)) {
                console.log(`需要删除${fullPath}`)
                delCount++;
                delFunc(fullPath, { recursive: true });
            }
        })
    });
}

console.log('开始清理')
clean(process.cwd())
console.log(`清理完毕，共删除${delCount}个文件`)

File: clean.py
import os
import shutil

ignore_files = {'.git', '.vscode', '.gitignore'}
remove_file_rule = ['cmake_install.cmake', 'CMakeCache.txt', 'CMakeFiles', 'Makefile', 'build']
remove_file_suffix = ['-bin']

del_count =0

def clean(dir_path):
    for file in os.listdir(dir_path):
        full_path = os.path.join(dir_path, file)

        if file in ignore_files:
            continue

        if os.path.isdir(full_path):
            clean(full_path)
            del_func = shutil.rmtree
        else:
            del_func = os.remove

        def delete_file():
            print(f'需要删除 {full_path}')
            del_count += 1
            del_func(full_path)

        for rule in remove_file_rule:
            if rule in file:
                delete_file()

        for suffix in remove_file_suffix:
            if file.endswith(suffix):
                delete_file()

print('开始清理')
clean(os.getcwd())
print(f'清理完毕，共删除 {del_count} 个文件')
